### PR TITLE
added detgate

### PR DESCRIPTION
**Pull request by PRAVEEN-mnl**

### CODE REVIEW

**Comment by PRAVEEN-mnl** [comment]
#4199

**Comment by PRAVEEN-mnl** [comment]
#4199 Put veto windows around detchar gates

File: bin/all_sky_search/pycbc_fit_sngls_split_binned [hunk 1]
```diff
@@ -238,6 +238,9 @@ if args.gating_veto_windows:
     if not (gveto_before == 0 and gveto_after == 0):
         time = trigf[args.ifo + '/end_time'][:]
         gate_times = np.unique(trigf[args.ifo + '/gating/auto/time'][:])
+        if args.ifo is not 'V1':
```
Comment (tdent) [inline bin/all_sky_search/pycbc_fit_sngls_split_binned:241]: We should not assume that H and L have file gates and V never has them, eg V1 could produce file gates in future. Instead, check if the ifo/gating/file/ group exists and if so read the times.

File: bin/all_sky_search/pycbc_fit_sngls_split_binned [hunk 2]
```diff
@@ -238,6 +238,9 @@ if args.gating_veto_windows:
...
+            detgate_times = np.unique(trigf[args.ifo + '/gating/file/time'][:])
+        gate_times.extend(detgate_times)
```
Comment (tdent) [inline bin/all_sky_search/pycbc_fit_sngls_split_binned:243]: noting that this should be inside the 'if' switch and use numpy,concatenate()

File: bin/all_sky_search/pycbc_fit_sngls_by_template [hunk 3]
```diff
@@ -212,7 +212,10 @@ if args.gating_veto_windows:
         raise ValueError("Gating veto window values must be negative before "
                          "gates and positive after gates.")
     if not (gveto_before == 0 and gveto_after == 0):
-        gate_times = np.unique(trigf[args.ifo + '/gating/auto/time'][:])
+        autogate_times = np.unique(trigf[args.ifo + '/gating/auto/time'][:])
+        if args.ifo+'/gating/file' in trigf:
+            detgate_times = trigf[args.ifo + '/gating/file/time'][:]
+            gate_times = np.concatenate((autogate_times, detgate_times))
         gveto_segs = events.veto.start_end_to_segments(gate_times + gveto_before,
```
Comment (veronica-villa) [inline bin/all_sky_search/pycbc_fit_sngls_by_template:219]: this is not going to work properly in the case of not having any detechar gates (code won't enter into the if loop and it won't know what `gate_times` is)
I think it's ok

```
autogate_times = np.unique(trigf[args.ifo + '/gating/auto/time'][:])
    if args.ifo + '/gating/file' in trigf:
        detgate_times = trigf[args.ifo + '/gating/file/time'][:]
    else:
        detgate_times = []   
gate_times = np.concatenate((autogate_times, detgate_times))

**Comment by tdent** [comment]
I did some minor fixes of formatting (spaces around ' + ' etc), should not affect the code functionality

**Review by tdent — APPROVED** [review]

**Pull request merged**